src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs

```diff
@@ -235,6 +235,52 @@ public async Task Pound_r_nuget_disallows_empty_package_specification(Language l
             .Be("(1,4): error DNI210: Unable to parse package reference: \"nuget:\"");
     }
 
+    [Fact]
+    public async Task Pound_r_is_not_treated_as_pound_r_nuget_by_csharp_kernel_if_assembly_path_happens_to_contain_the_string_nuget()
+    {
+        var kernel = CreateCSharpKernel();
+
+        using var events = kernel.KernelEvents.ToSubscribedList();
+
+        await kernel.SubmitCodeAsync("""
+                                     #r "C:\Users\abcde\.nuget\packages\package\1.0.0\package.dll"
+                                     """);
+
+        events
+            .Should()
+            .ContainSingle<CommandFailed>()
+            .Which
+            .Message
+            .Should()
+            .Contain("Metadata file")
+            .And
+            .Contain("could not be found")
+            .And
+            .NotContain("DNI");
+    }
+
+    [Fact]
+    public async Task Pound_r_is_not_treated_as_pound_r_nuget_by_fsharp_kernel_if_assembly_path_happens_to_contain_the_string_nuget()
+    {
+        var kernel = CreateKernel(Language.FSharp);
+
+        using var events = kernel.KernelEvents.ToSubscribedList();
```

Comment:
It's generally best to avoid testing using `kernel.KernelEvents` and instead check the `KernelCommandResult.Events` for the specific command you're sending.

Reply:
Fixed